### PR TITLE
Implement LWG-4001 `iota_view` should provide `empty`

### LINKED ISSUE #4159: LWG-4001 `iota_view` should provide `empty`
LWG-4001 `iota_view` should provide `empty`

### PR DESCRIPTION
**Pull request by frederick-vs-ja**
Fixes #4159.

### CODE REVIEW

File: stl/inc/ranges [hunk 1]
```diff
@@ -1399,6 +1399,10 @@ namespace ranges {
             }
         }
 
+        _NODISCARD_EMPTY_MEMBER constexpr bool empty() const noexcept(noexcept(_Value == _Bound)) /* strengthened */ {
```
Comment (StephanTLavavej) [inline stl/inc/ranges:1402]: `_NODISCARD_EMPTY_MEMBER` mentions that `clear()` may be intended, which is not a thing for `iota_view`:

https://github.com/microsoft/STL/blob/46843b3bf320c6070ff63e978936bcf898e5a2bc/stl/inc/yvals_core.h#L596-L600

Other views use plain `_NODISCARD`:

https://github.com/microsoft/STL/blob/46843b3bf320c6070ff63e978936bcf898e5a2bc/stl/inc/ranges#L1833

File: tests/std/tests/P0896R4_views_iota/test.cpp [hunk 2]
```diff
@@ -387,6 +406,25 @@ int main() {
...
 
         assert(ranges::equal(r, even_ints, ranges::equal_to{}, deref));
+
+        using FilteredIota = decltype(r);
+        static_assert(CanEmpty<FilteredIota>);
+        static_assert(CanMemberEmpty<FilteredIota>);
+    }
+    // LWG-4001 iota_view should provide empty
+    {
+        using BackInsertingIota = decltype(views::iota(back_inserter(declval<vector<int>&>())));
```
Comment (StephanTLavavej) [inline tests/std/tests/P0896R4_views_iota/test.cpp:416]: Should include `<iterator>` for `back_inserter()`. (\[iterator.range\] guarantees that `<vector>` drags in non-member `begin()` etc. but not the rest of `<iterator>`.)

**Comment by StephanTLavavej** [comment]
Thanks! I pushed minor changes, and I think this needs only one maintainer approval.

**Comment by StephanTLavavej** [comment]
:abacus: :eyeglasses: :zero:
===

File: tests/std/tests/P0896R4_views_iota/test.cpp (hunk 2 above)
Comment (CaseyCarter) [reply tests/std/tests/P0896R4_views_iota/test.cpp:417]: [`<ranges>` includes `<iterator>`](https://eel.is/c++draft/ranges.syn).

**Pull request merged**